src/internal/observable/defer.ts

```diff
@@ -52,7 +52,14 @@ import { from } from './from'; // lol
  * @owner Observable
  */
 export function defer<R extends ObservableInput<any>>(
-  observableFactory: [R] extends [undefined] | [void] ? never : () => R
+  observableFactory:
+    [R] extends [never] ?
+      never :
+      [R] extends [any] ?
+        () => R :
+        [R] extends [undefined] | [void] ?
+          never :
+          () => R
```

Comment:
I'm always suspicious that the simplest thing is probably the correct thing (however I'm frequently wrong, but this looks suspicious to me)...  Just to be sure we're clear here: What this is doing is saying "This has to be an observable input, such that: `void|undefined|never` are errors, and `any` will result in `Observable<unknown>`? Is that correct?

Is there any way to make a utility type out of this? Do we need this anywhere else? 🤔 

The returning of `any` resulting in an `Observable<unknown>` seems useful, however it sort of breaks `any`, right? IMO, `any` is the escape hatch people use when they just give up.  Let's say someone has a hard time figuring out the types to a defer that would otherwise return valid things:

```ts
defer(() => {
   return getSomethingAuthorIsTooLazyToType() as any
}).pipe(
   map((thing) => thing.methodThatIsAlwaysThere())
)
```

If this is `Observable<unknown>` we just broke `any` for them, and they have to kick the `any` down the way...

And _maybe that is what we want to do_... I'm just discussing at this point, not asking for a change.

@cartant ? @kwonoj ?

I'm truly on the fence about this one.

Reply:
@benlesh I thought the original problem was that `defer` accepted functions that returned `void` and I thought we'd agreed that the solution was to remove the overload signature that allowed that. I was surprised to see that the code this PR fixes was merged. See [my comment above](https://github.com/ReactiveX/rxjs/pull/5513#issuecomment-647073283).

Reply:
Yeah... so that was the original issue. Then, with the new typings that fixed that _original issue_, we caused issues with factories that returned `any`, which should be an escape hatch for developers and "just work"... because `any`. Which is why I filed the issue that this is trying to solve.

I'm just not sure returning `Observable<unknown>` is right here. Really, truly, on the fence about it.

Reply:
I think maybe there was a misunderstanding with your comment in the other PR, at least on my end. I thought you were talking about the runtime behavior, which was indeed resolved, but you were generally approving of the typings. Which looked fine to me. But those typings caused issues with `() => any` , which is what we're trying to resolve here.

Reply:
I'm saying that I don't think [this fix](https://github.com/ReactiveX/rxjs/pull/5449) was the best way of fixing the problem. And that this fix exacerbates it. It seems too complicated, IMO.

I think the fix should have been what's in [this comment](https://github.com/ReactiveX/rxjs/pull/5449#discussion_r428276385).

Reply:
Yeah. I agree.

Funny enough, the recommended fix has the same outcome. Was talking with @jayphelps about this offline, and he pointed this out.

```ts
declare function foo<T>(cb: () => T[]): T[] { }

const result = foo(() => 1 as any); // unknown[]
```

Reply:
I think is still needs the `ObservedValueOf` business to handle factories with multiple return paths of different types, but, yeah, I think the types and - the implementation - should be changed to require a factory that actually returns an observable input. That's easily the 95% use case, IMO. If someone wants to return `EMPTY`, they can do so explicitly.